XiEditor/XiTextPlane/TextLine.swift

```diff
@@ -168,6 +168,24 @@ struct TextLine {
     var width: Double {
         return CTLineGetTypographicBounds(ctLine, nil, nil, nil)
     }
+
+    func scatterGather<C: Collection>(indices glyphIndices: C, font: CTFont)
+            -> TextLine where C.Element == Int, C.IndexDistance == Int {
+        var newGlyphs = Array<GlyphInstance>()
+        newGlyphs.reserveCapacity(glyphIndices.count)
+
+        var glyphPosition = 0
+        for i in glyphIndices {
+            newGlyphs.append(self.glyphs[i])
+            newGlyphs[glyphPosition].x = self.glyphs[glyphPosition].x
+            newGlyphs[glyphPosition
+                ].y = self.glyphs[glyphPosition].y
+            glyphPosition += 1
+        }
+        let newWidth = self.width * (Double(glyphIndices.count)) / (Double(self.glyphs.count))
+        let newLine = CTLineCreateTruncatedLine(ctLine, newWidth, CTLineTruncationType.start, nil)!
```

Comment:
This unwrap is potentially trouble; there are legitimate instances where this function may return nil. As an example,  this will currently crash if the font is "Hoefler Text". (found by absolute accident)

This might be revealing some of the assumptions we're making around monospaced type; I do think it's reasonable that the gutter if visible always uses a mono typeface, but this isn't enforced anywhere. (Also this doesn't fail on _all_ fonts with variable-width numerals).